async/as/ThreadBase.h

```diff
@@ -104,19 +125,33 @@ class ThreadBase : public Base<Executor, InitParameter, Parameter>
 		  m_phase(EXEC_PHASE),
 		  m_shutdown(false)
 	{
-		pthread_spin_init(&m_writerLock, PTHREAD_PROCESS_PRIVATE);
-		pthread_mutex_init(&m_readerLock, 0L);
-		pthread_spin_init(&m_initBufferLock, PTHREAD_PROCESS_PRIVATE);
+#ifndef __APPLE__
+                pthread_spin_init(&m_writerLock, PTHREAD_PROCESS_PRIVATE);
+                pthread_spin_init(&m_initBufferLock, PTHREAD_PROCESS_PRIVATE);
+#else
+                pthread_mutexattr_t attr;
+                pthread_mutexattr_init(&attr);
+                pthread_mutexattr_settype(&attr, PTHREAD_PROCESS_PRIVATE);
+                pthread_mutex_init(&m_writerLock, &attr);
+                pthread_mutex_init(&m_initBufferLock, &attr);
+
+#endif // __APPLE__
+                pthread_mutex_init(&m_readerLock, 0L);
 	}
 
 public:
 	~ThreadBase()
 	{
 		finalize();
 
-		pthread_spin_destroy(&m_initBufferLock);
-		pthread_mutex_destroy(&m_readerLock);
-		pthread_spin_destroy(&m_writerLock);
+#ifndef __APPLE__
```

Comment:
It would be best to maybe (though it's not needed for this PR, ultimately) also put a distinction for these methods up to `spin_lock` etc.? Or create a lock class directly which incorporates all these methods.